contracts/multisig-prover/src/encoding/abi2/mod.rs

```diff
@@ -0,0 +1,205 @@
+use alloy_primitives::Address;
+use alloy_sol_types::{sol, SolValue};
+use cosmwasm_std::{HexBinary, Uint128, Uint256};
+use k256::{elliptic_curve::sec1::ToEncodedPoint, PublicKey};
+use sha3::{Digest, Keccak256};
+
+use axelar_wasm_std::hash::Hash;
+use multisig::{key::PublicKey as MultisigPublicKey, msg::Signer, worker_set::WorkerSet};
+
+use crate::{error::ContractError, payload::Payload};
+
+sol!("src/encoding/abi2/solidity/AmplifierGatewayTypes.sol");
+sol!("src/encoding/abi2/solidity/WeightedMultisigTypes.sol");
+
+impl From<&Payload> for CommandType {
+    fn from(payload: &Payload) -> Self {
+        match payload {
+            Payload::Messages(_) => CommandType::ApproveMessages,
+            Payload::WorkerSet(_) => CommandType::RotateSigners,
+        }
+    }
+}
+
+impl WeightedSigners {
+    pub fn hash(&self) -> Hash {
+        Keccak256::digest(self.abi_encode()).into()
+    }
+}
+
+impl From<&Signer> for WeightedSigner {
+    fn from(signer: &Signer) -> Self {
+        let weight: Uint128 = signer
+            .weight
```

Comment:
not necessarily for this PR, but we should just change the signer weight to uint128